test/optimizer/stats_storage_test.cpp

```diff
@@ -1,58 +1,116 @@
 #include "optimizer/statistics/stats_storage.h"
 
-#include <utility>
-
+#include "catalog/database_catalog.h"
 #include "gtest/gtest.h"
+#include "storage/sql_table.h"
+#include "test_util/end_to_end_test.h"
 #include "test_util/test_harness.h"
 
 namespace noisepage::optimizer {
-class StatsStorageTests : public TerrierTest {
+class StatsStorageTests : public test::EndToEndTest {
```

Comment:
Actually, is it possible to add a test that modifies the `StatsStorage` concurrently? Since there're two layers of latches and we've been back-and-forth on this a little bit, I feel like it would be good to have a test case for it.

Reply:
Added